game/src/main/java/org/apollo/game/model/entity/EntityBounds.java

```diff
@@ -0,0 +1,89 @@
+package org.apollo.game.model.entity;
+
+import org.apollo.game.model.Direction;
+import org.apollo.game.model.Position;
+
+/**
+ * The bounds of an {@link Entity}.
+ *
+ * @author Steve Soltys
+ */
+public class EntityBounds {
+
+	/**
+	 * The {@link Entity}.
+	 */
+	private final Entity entity;
+
+	/**
+	 * The last known Position of the Entity.
+	 */
+	private Position position = new Position(0, 0);
+
+	/**
+	 * The last known width of the Entity.
+	 */
+	private int width = 0;
+
+	/**
+	 * The last known height of the Entity.
+	 */
+	private int height = 0;
+
+	/**
+	 * The set of positions that the Entity occupies.
+	 */
+	private Position[] positions = new Position[0];
+
+	/**
+	 * Creates an EntityBounds.
+	 *
+	 * @param entity The entity.
+	 */
+	EntityBounds(Entity entity) {
+		this.entity = entity;
+	}
+
+	/**
+	 * Checks whether the given position is within the Entity's bounds.
+	 *
+	 * @param position The position.
+	 * @return A flag indicating whether or not the position exists within the Entity's bounds.
+	 */
+	public boolean contains(Position position) {
+		Position entityPosition = entity.getPosition();
+		int width = entity.getWidth();
+		int height = entity.getLength();
+
+		return position.getX() >= entityPosition.getX() && position.getX() < entityPosition.getX() + width &&
+			position.getY() >= entityPosition.getY() && position.getY() < entityPosition.getY() + height &&
+			position.getHeight() == entityPosition.getHeight();
+	}
+
+	/**
+	 * Gets the set of positions that the Entity occupies.
+	 * <p>
+	 * If the width, height, or Position of the entity has changed since the last call, it will calculate a new
+	 * positions set before returning.
+	 *
+	 * @return The set of positions.
+	 */
+	public Position[] getPositions() {
+
+		if (!position.equals(entity.getPosition()) || width != entity.getWidth() || height != entity.getLength()) {
```

Comment:
Is it better to just calculate these positions on each call since we have a contains function now, or should we keep it this way?

Reply:
I'm not sure if this is any better than your original PR, just looks decoupled...

Reply:
I think I'm misunderstanding the reasoning for keeping all of the `Position`s handy rather than just keeping the area that we occupy

Reply:
Even if it isn't better than before, it's certainly not the same logic.. We also have `contains` now for checking if a position exists within the bounds, which satisfies the use case for the rectangle you're suggesting (it's really a polygon since width and length can be different).

This function is for getting the actual set of positions that the `Entity` occupies. The change is that it takes @garyttierney's suggestion of not creating a temporary object every call into account, but I'm not sure which is preferable now that we have `contains`.

If you're asking why this function exists, it's because I use it in a new distanced action implementation that I plan on merging later.

Reply:
Agree with what Ryley said.  Where would the result of this call be used?

Reply:
If you prefer, we can leave this out, but it's possible we may need to merge something like it in later.

Reply:
Let's leave it out for now.  I'm not convinced it's needed in the distanced action implementation, so can discuss that when we come to it.

Reply:
I'm just trying to think of an application where this API would deem useful. If you've already written something or have an idea, do share. :)

Reply:
Also, can't only `GameObject`'s be considered a polygon? Player/Npc/Projectile/GroundItem are always the same width and length, correct? Maybe the implementation should be altered only for `GameObject` if this is the case..

Reply:
@ryleykimmel sure, I gotcha. Hop on slack when you get a chance, we had a small bit of discussion there.